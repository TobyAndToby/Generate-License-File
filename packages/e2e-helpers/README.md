# e2e-helpers

This library was generated with [Nx](https://nx.dev).

## Building

Run `nx build e2e-helpers` to build the library.
