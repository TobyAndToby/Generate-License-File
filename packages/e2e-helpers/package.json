{
  "name": "@generate-license-file/e2e-helpers",
  "version": "0.0.1",
  "type": "commonjs"
}
