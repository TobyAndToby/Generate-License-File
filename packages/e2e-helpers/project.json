{
  "name": "e2e-helpers",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/e2e-helpers/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/e2e-helpers",
        "main": "packages/e2e-helpers/src/index.ts",
        "tsConfig": "packages/e2e-helpers/tsconfig.lib.json",
        "assets": ["packages/e2e-helpers/*.md"],
        "buildableProjectDepsInPackageJsonType": "dependencies"
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/e2e-helpers/**/*.ts"]
      }
    }
  },
  "tags": []
}
