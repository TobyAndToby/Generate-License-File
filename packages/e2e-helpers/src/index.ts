export * from "./lib/lineEndings";
export * from "./lib/pathTypes";
