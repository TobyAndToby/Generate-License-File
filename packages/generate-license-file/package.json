{
  "name": "generate-license-file",
  "version": "0.0.0",
  "description": "Generates a text file containing all of the licenses for your production dependencies",
  "main": "src/index.js",
  "license": "ISC",
  "type": "commonjs",
  "contributors": [
    {
      "name": "Toby Smith",
      "url": "https://github.com/tobysmith568"
    },
    {
      "name": "Toby Bessant",
      "url": "https://github.com/tobybessant"
    }
  ],
  "files": [
    "src",
    "bin"
  ],
  "keywords": [
    "cli",
    "license",
    "licence",
    "third",
    "party",
    "file",
    "generate",
    "oss"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/TobyAndToby/generate-license-file.git"
  },
  "bugs": {
    "url": "https://github.com/TobyAndToby/generate-license-file/issues"
  },
  "homepage": "https://generate-license-file.js.org",
  "bin": {
    "generate-license-file": "bin/generate-license-file"
  }
}
