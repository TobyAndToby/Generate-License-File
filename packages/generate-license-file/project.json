{
  "name": "generate-license-file",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/generate-license-file/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/generate-license-file",
        "main": "packages/generate-license-file/src/index.ts",
        "tsConfig": "packages/generate-license-file/tsconfig.lib.json",
        "assets": [
          "packages/generate-license-file/*.md",
          "packages/generate-license-file/bin/**",
          "LICENSE.md"
        ],
        "buildableProjectDepsInPackageJsonType": "dependencies"
      }
    },
    "run": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node ./dist/packages/generate-license-file/bin/generate-license-file"
      }
    },
    "version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/version.mjs generate-license-file {args.version}"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/publish.mjs generate-license-file {args.tag}"
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/generate-license-file/**/*.ts"],
        "maxWarnings": 0
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "packages/generate-license-file/jest.config.ts",
        "passWithNoTests": false
      }
    }
  },
  "tags": []
}
