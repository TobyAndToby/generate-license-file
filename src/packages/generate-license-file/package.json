{
  "name": "generate-license-file",
  "version": "*",
  "description": "Generates a text file containing all of the licenses for your production dependencies",
  "main": "src/index.js",
  "license": "ISC",
  "type": "commonjs",
  "contributors": [
    {
      "name": "Toby Smith",
      "url": "https://github.com/tobysmith568"
    },
    {
      "name": "Toby Bessant",
      "url": "https://github.com/tobybessant"
    }
  ],
  "files": [
    "src",
    "bin"
  ],
  "keywords": [
    "cli",
    "license",
    "licence",
    "third",
    "party",
    "file",
    "generate",
    "oss"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/TobyAndToby/generate-license-file.git",
    "directory": "src/packages/generate-license-file"
  },
  "bugs": {
    "url": "https://github.com/TobyAndToby/generate-license-file/issues"
  },
  "homepage": "https://generate-license-file.js.org",
  "bin": {
    "generate-license-file": "bin/generate-license-file"
  },
  "dependencies": {
    "@commander-js/extra-typings": "^13.1.0",
    "@npmcli/arborist": "^8.0.0",
    "cli-spinners": "^2.6.0",
    "commander": "^13.1.0",
    "cosmiconfig": "^9.0.0",
    "enquirer": "^2.3.6",
    "json5": "^2.2.3",
    "ora": "^5.4.1",
    "tslib": "^2.3.0",
    "zod": "^3.21.4",
    "glob": "^10.4.5"
  }
}
