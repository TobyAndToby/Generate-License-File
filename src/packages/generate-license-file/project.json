{
  "name": "generate-license-file",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/generate-license-file/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/generate-license-file/**/*.ts"],
        "maxWarnings": 0
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/generate-license-file",
        "main": "packages/generate-license-file/src/index.ts",
        "tsConfig": "packages/generate-license-file/tsconfig.lib.json",
        "assets": [
          "packages/generate-license-file/*.md",
          "packages/generate-license-file/bin/**",
          "LICENSE.md"
        ],
        "buildableProjectDepsInPackageJsonType": "dependencies"
      },
      "dependsOn": ["^build"]
    },
    "run": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node ./dist/packages/generate-license-file/bin/generate-license-file"
      },
      "dependsOn": ["build"]
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "packages/generate-license-file/jest.config.ts",
        "passWithNoTests": false
      }
    },
    "version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/version.mjs generate-license-file {args.version}"
      }
    },
    "pack": {
      "executor": "nx:run-commands",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packs/generate-license-file",
        "command": "node tools/scripts/pack.mjs generate-license-file"
      },
      "dependsOn": ["build"]
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/publish.mjs generate-license-file {args.tag}"
      }
      // "dependsOn": ["pack"]
    }
  },
  "tags": []
}
