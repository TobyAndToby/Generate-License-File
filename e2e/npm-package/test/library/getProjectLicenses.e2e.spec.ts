import { describeRelativeAndAbsolutePaths } from "@generate-license-file/e2e-helpers";
import { getProjectLicenses } from "generate-license-file";

describe("getProjectLicenses", () => {
  describeRelativeAndAbsolutePaths("./package.json", (packageJsonPath) => {
    it("should match snapshot", async () => {
      const result = await getProjectLicenses(packageJsonPath);

      expect(result).toMatchSnapshot();
    });
  });

  it("should not include versions when omitVersions is true", async () => {
    const result = await getProjectLicenses("./package.json", {
      omitVersions: true,
    });

    expect(result).toMatchSnapshot();
  });
});
