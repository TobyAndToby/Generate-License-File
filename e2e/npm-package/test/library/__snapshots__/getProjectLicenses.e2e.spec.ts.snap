// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProjectLicenses for a relative path should match snapshot 1`] = `
[
  {
    "content": "MIT",
    "dependencies": [
      "dep-five@1.0.0",
    ],
  },
  {
    "content": "# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.",
    "dependencies": [
      "dep-four@1.0.0",
    ],
  },
  {
    "content": "# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.
",
    "dependencies": [
      "dep-one@1.0.0",
    ],
  },
  {
    "content": "# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.",
    "dependencies": [
      "dep-three@1.0.0",
    ],
  },
  {
    "content": "# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.
",
    "dependencies": [
      "dep-two@1.0.0",
      "dep-two-duplicate@1.0.0",
    ],
  },
]
`;

exports[`getProjectLicenses for an absolute path should match snapshot 1`] = `
[
  {
    "content": "MIT",
    "dependencies": [
      "dep-five@1.0.0",
    ],
  },
  {
    "content": "# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.",
    "dependencies": [
      "dep-four@1.0.0",
    ],
  },
  {
    "content": "# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.
",
    "dependencies": [
      "dep-one@1.0.0",
    ],
  },
  {
    "content": "# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.",
    "dependencies": [
      "dep-three@1.0.0",
    ],
  },
  {
    "content": "# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.
",
    "dependencies": [
      "dep-two@1.0.0",
      "dep-two-duplicate@1.0.0",
    ],
  },
]
`;
